### ISSUE TITLE
 RecyclableMemoryStream allocates  repeatedly the  available RAM ?

### ISSUE THREAD

**Issue by stefan2410** [microsoft/Microsoft.IO.RecyclableMemoryStream#175]
Version 2.1.1
Suddenly,  it starts to allocate all available RAM, till less than 1 GB free, freeing and trying to reallocate it again, repeatedly.
e.g. Available RAM in 36 min. (3 min intervals):  5952  6109  1841  5962  3744  1205  4434  5276  3238  1753  7732  6720 
There are not so many streams allocations, neither a stream longer .500MB,    usually less than  50 MB

**Comment by stefan2410** [comment]
It seems that with version 2.1.2 I don't have this problem

**Comment by benmwatson** [comment]
2.1.2 fixed a problem with an integer overflow that could cause infinite loops, but that problem existed since before 2.1.1. Could you put together a little repro program?

**Comment by stefan2410** [comment]
@benmwatson  I can't reproduce it as it runs in production servers.  You are right  that "the problem existed since before 2.1.1". Simply I used the version before 2.0  for a year till 10-15 of May, when I upgraded it to 2.0. and then 2.1.
The configuration was unbounded (no MaximumFreeSmallPoolBytes) because I can control the streams, having common characteristics.  When I tried in ver 2+  to put limits I got worst behavior but I had the random problem ocassionally without limits. 
With ver 2.1.2 it works fine now and with limits, estimated workload.
I'm suspicious that the problem was appeared because t I asked regularely streams with  initial capacity larger than the small pool block size. I changed it now also.
